src/main.c

```diff
@@ -101,6 +100,112 @@ static void init_sdl(void) {
 	log_info("Using SDL %u.%u.%u", v.major, v.minor, v.patch);
 }
 
+static char* get_default_res_path(void) {
+	char *res;
+
+#ifdef RELATIVE
+	res = SDL_GetBasePath();
+	strappend(&res, "data/");
+#else
+	res = strdup(STATIC_RESOURCE_PREFIX);
+#endif
+
+	return res;
+}
+
+static void get_core_paths(char **res, char **storage) {
+	if((*res = getenv("TAISEI_RES_PATH")) && **res) {
+		*res = strdup(*res);
+	} else {
+		*res = get_default_res_path();
+	}
+
+	if((*storage = getenv("TAISEI_STORAGE_PATH")) && **storage) {
+		*storage = strdup(*storage);
+	} else {
+		*storage = SDL_GetPrefPath("", "taisei");
+	}
+}
+
+static bool filter_zip_ext(const char *str) {
+	char buf[strlen(str) + 1];
+	memset(buf, 0, sizeof(buf));
+
+	for(char *p = buf; *str; ++p, ++str) {
+		*p = tolower(*str);
+	}
+
+	return strendswith(buf, ".zip");
+}
+
+static void init_vfs(bool silent) {
+	char *res_path, *storage_path;
+	get_core_paths(&res_path, &storage_path);
+
+	if(!silent) {
+		log_info("Resource path: %s", res_path);
+		log_info("Storage path: %s", storage_path);
+	}
+
+	char *p = NULL;
+
+	struct mpoint_t {
+		const char *dest;	const char *syspath;							bool mkdir;
+	} mpts[] = {
+		{"storage",			storage_path,									true},
+		{"resdirs",			res_path,										false},
+		{"resdirs",			p = strfmt("%s/resources", storage_path),		true},
+		{NULL}
+	};
+
+	vfs_init();
```

Comment:
@laochailan I think it's a naming issue. vfs_init is part of the VFS API, it sets up the root and  other bare minimum things required to use the VFS in any way. init_vfs is a consumer function, it calls vfs_init and configures the VFS for use in Taisei. So maybe something like "setup_vfs" would be a more appropriate name for it?